### PR TITLE
Fix Large VFP-relative memory-references

### PR DESCRIPTION
**Pull request by BradleyWood**
See my comments on the issue [here](https://github.com/eclipse-openj9/openj9/issues/15363#issuecomment-1481820466).

Fixes eclipse-openj9/openj9#15363

### CODE REVIEW

**Comment by BradleyWood** [comment]
jenkins build all

**Comment by BradleyWood** [comment]
jenkins build all

**Comment by BradleyWood** [comment]
@0xdaryl Can you review?

**Comment by 0xdaryl** [comment]
The comments here are minor (I think).  Can you address so we can get this in?  Thanks.

File: compiler/x/amd64/codegen/OMRMemoryReference.cpp [hunk 1]
```diff
@@ -190,6 +190,12 @@ void OMR::X86::AMD64::MemoryReference::finishInitialization(
       {
       mightNeedAddressRegister = true;
       }
+   else if (sr.getSymbol() != NULL && (sr.isUnresolved() || (sr.stackAllocatedArrayAccess() && !IS_32BIT_SIGNED(self()->getDisplacement()))))
```
[the remark this answers is not in the document]
Comment (BradleyWood) [reply compiler/x/amd64/codegen/OMRMemoryReference.cpp:193]: Yes, that's right.

**Comment by 0xdaryl** [comment]
Jenkins build xlinux,win,osx,x32linux

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**